src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -252,6 +254,54 @@ private function validateAndDenormalize($currentClass, $attribute, $data, $forma
                 $data = [$data];
             }
 
+            // In XML and CSV all basic datatypes are represented as strings, it is e.g. not possible to determine,
```

Comment:
For XML, isn't it better to do XSD validation in the encoder instead?

For CSV, all bets are off. We can't reasonably know how to convert anything here.

Reply:
Would that mean that developers would need to create an XSD document for each and every entity they want to deserialize in order to work with XML (having already defined type information separately in the Entity)? Is that practical? XSD would be a nice feature for validation but at the moment XML deserialization is essentially unusable for any data that includes floats and integers and there aught to be a way of dealing with that. XML APIs will typically interpret values set on boolean properties as booleans. Validation is a separate concern to that.